File: lessons/gen_scripts/php/scripts/o00.flags.php
#!/usr/bin/env php
<?php
// PHP's getopt() doesn't reliably recognize "?" as an option character
//  (it silently finds nothing for -?, even with "?" in the optstring) -
//  so -? is detected with a plain in_array() check against $argv instead.
$usage = "\nUsage: {$argv[0]} [-c|-e|-l|-k|-p|-m|-t] [-h|-?]\n\n" .
         "  -c  Coffee\n" .
         "  -e  Espresso\n" .
         "  -l  Latte\n" .
         "  -k  Machiato\n" .
         "  -p  Capucino\n" .
         "  -m  Mocha\n" .
         "  -t  Tea\n" .
         "  -h  Display this help message\n" .
         "  -?  Display this help message\n\n";

$opts = getopt("celkpmth");

if (isset($opts["c"])) { echo "You ordered a Coffee.\n"; exit(0); }
if (isset($opts["e"])) { echo "You ordered an Espresso.\n"; exit(0); }
if (isset($opts["l"])) { echo "You ordered a Latte.\n"; exit(0); }
if (isset($opts["k"])) { echo "You ordered a Machiato.\n"; exit(0); }
if (isset($opts["p"])) { echo "You ordered a Capucino.\n"; exit(0); }
if (isset($opts["m"])) { echo "You ordered a Mocha.\n"; exit(0); }
if (isset($opts["t"])) { echo "You ordered a Tea.\n"; exit(0); }
if (isset($opts["h"]) || in_array("-?", $argv)) { echo $usage; exit(0); }

fwrite(STDERR, $usage);
exit(1);
?>

File: lessons/gen_scripts/perl/scripts/o00.flags.pl
#!/usr/bin/env -S perl -w
use Getopt::Std;

my $usage = <<END;

Usage: $0 [-c|-e|-l|-k|-p|-m|-t] [-h|-?]

  -c  Coffee
  -e  Espresso
  -l  Latte
  -k  Machiato
  -p  Capucino
  -m  Mocha
  -t  Tea
  -h  Display this help message
  -?  Display this help message

END

my %opts;
getopts('celkpmth?', \%opts);

if ($opts{c}) { print "You ordered a Coffee.\n"; exit 0; }
if ($opts{e}) { print "You ordered an Espresso.\n"; exit 0; }
if ($opts{l}) { print "You ordered a Latte.\n"; exit 0; }
if ($opts{k}) { print "You ordered a Machiato.\n"; exit 0; }
if ($opts{p}) { print "You ordered a Capucino.\n"; exit 0; }
if ($opts{m}) { print "You ordered a Mocha.\n"; exit 0; }
if ($opts{t}) { print "You ordered a Tea.\n"; exit 0; }
if ($opts{h} || $opts{'?'}) { print $usage; exit 0; }

print STDERR $usage;
exit 1;
